Framework/Geometry/src/Objects/BoundingBox.cpp

```diff
@@ -65,60 +65,60 @@ bool BoundingBox::doesLineIntersect(const V3D &startPoint,
   if (startPoint.X() > xMax()) {
     if (lineDir.X() < -tol) {
       lambda = (xMax() - startPoint.X()) / lineDir.X();
-      if (yMin() < startPoint.Y() + lambda * lineDir.Y() &&
-          yMax() > startPoint.Y() + lambda * lineDir.Y())
-        if (zMin() < startPoint.Z() + lambda * lineDir.Z() &&
-            zMax() > startPoint.Z() + lambda * lineDir.Z())
+      if (yMin() <= startPoint.Y() + lambda * lineDir.Y() &&
```

Comment:
Can I ask why it was necessary to change this calculation?

Reply:
I found it failed when the ray enters the bounding box at one edge and left the bounding box at the opposite edge. Equality is necessary to catch this edge case.

Reply:
I've also added a unit test for this case.

Reply:
I see. Thanks for the explanation.